core/appeal/service.go

```diff
@@ -1017,7 +1019,18 @@ func (s *Service) addCreatorDetails(a *domain.Appeal, p *domain.Policy) error {
 			if p.IAM.Schema != nil {
 				creator = map[string]interface{}{}
 				for schemaKey, targetKey := range p.IAM.Schema {
-					creator[schemaKey] = userDetailsMap[targetKey]
+					if strings.Contains(targetKey, "$response") {
+						params := map[string]interface{}{
+							"response": userDetailsMap,
+						}
+						v, err := evaluator.Expression(targetKey).EvaluateWithVars(params)
+						if err != nil {
+							return fmt.Errorf("evaluating expression: %w", err)
+						}
+						creator[schemaKey] = v
+					} else {
+						creator[schemaKey] = userDetailsMap[targetKey]
+					}
```

Comment:
Can we simplify this block 😄 , the level of nesting in this block starting from `if userDetailsMap, ok := userDetails.(map[string]interface{}); ok` check is quite high. Maybe return early in starting flow?